pep-9999.txt

```diff
@@ -0,0 +1,212 @@
+interest has prompted a re-analysis and therefore this PEP.
+
+Background
+==========
+
+PEP-0472 was opened in 2014. The PEP focused on various use cases and was extracted
+from a broad discussion on implementation strategies. The PEP was eventually rejected
+in 2019 [#rejection]_ mostly due to lack of interest despite its 5 years of existence.
+
+However, with the introduction of type hints in PEP-0484 [#pep-0484]_ the
+square bracket notation has been used consistently to enrich the typing
+annotations, e.g. to specify a list of integers as Sequence[int]. As a result,
+a renewed interest in a more flexible syntax that would allow for named
+information has been expressed in many different threads.
+
+During the investigation of PEP-0472, many different strategies have been
+proposed to expand the language, but no real consensus was reached. Many corner
+cases have been examined more closely and felt awkward, backward incompatible
+or both. Renewed interest was prompted by Caleb Donovick [#request-1]_ in 2019 
+and Andras Tantos [#request-2]_ in 2020. These requests prompted a strong activity
+on the python-ideas mailing list, where various options have been discussed and
+a general consensus has been reached.
+
+Motivation and Use cases
+========================
+
+The current python syntax focuses exclusively on position to express the
+index, nd also contains syntactic sugar to refer to
+non-punctiform selection (slices)
+
+::
+
+    >>> a[3]       # returns the fourth element of a
+    >>> a[1:10:2]  # slice notation (extract a non-trivial data subset)
+    >>> a[3,2]     # multiple indexes (for multidimensional arrays)
+
+The additional notation proposed in this PEP would allow notations involving
+keyword arguments in the indexing operation, e.g.
+
+::
+
+    >>> a[K=3, R=2]
+
+which would allow a more flexible way to indicise content.
+
+One must additionally consider the extended form that allows both positional
+and keyword specification
+
+::
+
+    >>> a[3, R=3, K=4]
+
+Use cases
+---------
+
+The following practical use cases present two broad categories of usage of a
+keyworded specification: Indexing and contextual option. For indexing:
+
+1. To provide a more communicative meaning to the index, preventing e.g. accidental
+   inversion of indexes
+
+   ::
+
+     >>> gridValues[x=3, y=5, z=8]
+     >>> rain[time=0:12, location=location]
+
+2. To enrich the typing notation with keywords, especially during the use of generics
+
+   :: 
+
+     def function(value: MyType[T=int]):
+
+
+3. In some domain, such as computational physics and chemistry, the use of a
+   notation such as ``Basis[Z=5]`` is a Domain Specific Language notation to represent
+   a level of accuracy
+
+   ::
+
+     >>> low_accuracy_energy = computeEnergy(molecule, BasisSet[Z=3])
+
+4. Pandas currently uses a notation such as
+
+   ::
+    
+     >>> db[db['x'] == 1]
+
+   which could be replaced with db[x=1].
```

Comment:
It is uncommon to use the name `db` for a DataFrame.  `df` is far more common and would help the example.